applications/HDF5Application/python_scripts/point_set_output_process.py

```diff
@@ -0,0 +1,160 @@
+import KratosMultiphysics
+import KratosMultiphysics.HDF5Application as HDF5Application
+
+
+def Factory(parameters: KratosMultiphysics.Parameters,
+            model: KratosMultiphysics.Model):
+    if not isinstance(parameters, KratosMultiphysics.Parameters):
+        raise Exception("expecting input parameters of type KratosMultiphysics.Parameters, but got {}".format(type(parameters)))
+    return PointSetOutputProcess(model, parameters)
+
+
+class StepPattern:
+
+    def __init__(self,
+                 pattern: str,
+                 time_format = "",
+                 step_format = ""):
+        self.pattern = pattern
+        self.time_format = time_format
+        self.step_format = step_format
+
+        self.time_tag = "<time>"
+        self.step_tag = "<step>"
+
+    
+    def Substitute(self, model_part: KratosMultiphysics.ModelPart):
+        string = self.pattern
+
+        if self.time_tag in string:
+            time = model_part.ProcessInfo[KratosMultiphysics.TIME]
+            string = string.replace(self.time_tag, format(time, self.time_format))
+
+        if self.step_tag in string:
+            step = model_part.ProcessInfo[KratosMultiphysics.STEP]
+            string = string.replace(self.step_tag, format(step, self.step_format))
+
+        return string
+
+
+
+class PointSetOutputProcess(KratosMultiphysics.Process):
```

Comment:
```suggestion
class PointSetOutputProcess(KratosMultiphysics.OutputProcess):
```

This requires a bit of work, but otherwise we will need to refactor even more :/

see #7877

Reply:
Got it, makes sense. I just haven't seen ```OutputProcess``` before.

Reply:
> see #7877

Maybe this is something to put on the list. I am not sure about the plans to unify/update all output-type processes, but maybe after the KTC election #9093 this should be clarified.

Reply:
this was clarified, all processes that produce output should derive from `OutputProcess`
This was not done everywhere yet because
- It will be a breaking change
- it is work (it's easy if you do it :wink:)

hence new processes should definitively derive from `OutputProcess`

Reply:
> this was clarified, all processes that produce output should derive from `OutputProcess` This was not done everywhere yet because
> 
> * It will be a breaking change
> * it is work (it's easy if you do it 😉)
> 
> hence new processes should definitively derive from `OutputProcess`

Even it will break stuff, I think it should be high prority to make such updates. Going foward will just make dependencies more complicated. But I am not the one to decide in general.

This application should definitely align with this requirement, perhaps we can have a makeover of the HDFApp in this sense? @sunethwarna

Reply:
It is in my todo list for `HDF5Application`, along with changing to `parallel_utilities`. :)